### PR TITLE
Allow sorting by `$natural`

### LINKED ISSUE #3638: Allow sorting by `$natural`
We should support sorting by `{$natural: 1}` and `{$natural: -1}` by RecordID for capped collections only. That sorting can be pushed down safely.

See https://www.mongodb.com/docs/manual/reference/operator/meta/natural/. Note that `$natural` is field in `find's` `sort`. See  example below.

For non-capped collections, that should return an error.

The backend part is already done in #3793; we should update the handler and tests.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #3638.

CI failing due to coveralls.

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handler/msg_find.go [hunk 1]
```diff
@@ -216,6 +216,19 @@ func (h *Handler) makeFindQueryParams(params *common.FindParams, cInfo *backends
...
+		if params.Sort.Keys()[0] != "$natural" {
+			break
+		}
+
+		if !cInfo.Capped() {
+			return nil, handlererrors.NewCommandErrorMsgWithArgument(
+				handlererrors.ErrNotImplemented,
+				"$natural sort for non-capped collection is not supported.",
+				"find")
+		}
```
Comment (noisersup) [inline internal/handler/msg_find.go:229]: Left `NotImplemented` here, as apparently it's possible to run `$natural` against non-capped collection on mongo

File: integration/query_compat_test.go [hunk 2]
```diff
@@ -264,12 +265,45 @@ func TestQueryCappedCollectionCompat(t *testing.T) {
...
+		"SortNaturalString": {
+			skip:         "https://github.com/FerretDB/FerretDB/issues/3638",
+			sort:         bson.D{{"$natural", "foo"}},
+			sortPushdown: noPushdown,
+		},
+		"SortNaturalMultipleSorts": {
+			skip:         "https://github.com/FerretDB/FerretDB/issues/3638",
+			sort:         bson.D{{"$natural", int32(1)}, {"v", int32(1)}},
+			sortPushdown: noPushdown,
+		},
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply integration/query_compat_test.go:296]: Let's keep as it is for now

**Review by AlekSi — APPROVED** [review]

File: internal/handler/msg_find.go (hunk 1 above)
Comment (AlekSi) [reply internal/handler/msg_find.go]: It makes sense in some use cases, but yeah, let's not support it for now

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Uh…

CleanShot 2023-12-07 at 22 26 43@2x

File: integration/query_compat_test.go [hunk 3]
```diff
@@ -241,6 +241,7 @@ func TestQueryCappedCollectionCompat(t *testing.T) {
 	for name, tc := range map[string]struct {
```
Comment (AlekSi) [inline integration/query_compat_test.go:241]: Dummy conversation…

**Pull request merged**